python_modules/libraries/dagster-dbt/dagster_dbt/core/resources_v2.py

```diff
@@ -697,7 +697,9 @@ def stream(
                 def my_dbt_assets(context, dbt: DbtCliResource):
                     yield from dbt.cli(["run"], context=context).stream()
         """
-        return DbtEventIterator(self._stream_asset_events(), self)
+        return DbtEventIterator(
+            self._stream_asset_events(), self, ThreadPoolExecutor(STREAM_EVENTS_THREADPOOL_SIZE)
```

Comment:
- We should ensure that the threadpool's context is properly handled.
- `STREAM_EVENTS_THREADPOOL_SIZE` should potentially be modifiable by the user. Perhaps it should be set similar to `termination_timeout_seconds`, so that it can be overridden if needed.

Reply:
Updated to a `num_threads` param on each call which gives users the ability to control how much fan-out they want at each step & avoids us having to thread through the threadpool to each iterator, where managing lifecycle can be tricky. Since the number of chained calls is small, the overhead from opening a new pool a few more times should be minimal.